File: Data Structures & Algorithms/max-water-container/submission-3.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        currMax = -1
        l, r = 0, len(heights) - 1

        while l < r:
            height = min(heights[l], heights[r])
            dist = r - l
            amt = dist * height
            currMax = max(amt, currMax)
            if heights[l] < heights[r]:
                l += 1
            else:
                r -= 1

        return currMax

File: Data Structures & Algorithms/maximum-subarray/submission-3.java
class Solution {
    public int maxSubArray(int[] nums) {
        int currMax = nums[0];
        int curr = 0;

        for (int i = 0; i < nums.length; i++) {
            curr = Math.max(curr, 0);

            curr += nums[i];
            currMax = Math.max(currMax, curr);
        }

        return currMax;
    }
}
